src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -1466,6 +1466,7 @@ MarkRanges MergeTreeDataSelectExecutor::markRangesFromPKRange(
     }
 
     size_t used_key_size = key_condition.getMaxKeyColumn() + 1;
+    String part_name = part->isProjectionPart() ? fmt::format("{}.{}", part->name, part->getParentPart()->name) : part->name;
```

Comment:
Use `const String &` instead. It will extend the lifetime of rvalues and avoid unnecessary copy.